buildSrc/src/main/kotlin/kotest-jvm-conventions.gradle.kts

```diff
@@ -20,7 +20,7 @@ kotlin {
    sourceSets {
       val jvmTest by getting {
          dependencies {
-            implementation(project(Projects.JunitRunner))
+            implementation(project(":kotest-runner:kotest-runner-junit5"))
```

Comment:
The type-safe project accessors aren't available in buildSrc, so the project has to be referenced manually here.

Reply:
Makes sense. Would it be possible to solve this by moving buildSrc -> build-logic and using it as an included build instead? From what I can tell, that's the recommended way to handle build logic moving forward?

Reply:
Hmm no I don't think so, the generated project accessors are only available within a single Gradle build (i.e. there's a root `settings.gradle.kts`), not across included builds. It might be possible in other ways (moving `:kotest-runner-junit5` to an included build, and doing some dependency substitution, maybe?), but nothing that will be as simple as what's in this PR :)

On buildSrc... Normally I think buildSrc is fine. There's a drawback (every change in buildSrc requires a full rebuild), but that's not really such a big deal. The config in buildSrc doesn't change that much, and even when it does, it's not a heavy hit to re-build.

However, I think setting up an included build for the build plugins would help Kotest. It's a much larger project than most, and there are a lot more modules. Included builds would also help with #3171.

As a replacement for buildSrc, I'd probably set up two included builds. One for 'settings plugins'. These would be for plugins like version catalog, shared repository definitions, version alignment. These plugins would be used in all other projects, including the other included build, which would be more of a direct replacement for buildSrc, where you'd define convention plugins (for Kotlin, publishing, verification plugins, etc). 

And after that, once the ball is rolling, you could even include the Kotest extension projects as an included build. That would help with alignment, and sharing resources - but I'm just throwing out ideas.